app/models/availability.rb

```diff
@@ -21,4 +23,16 @@ def partially_available?(event)
     return false if event.nil? || (event.end_time == self.end_time && event.start_time == self.start_time)
     (event.end_time >= self.end_time && self.end_time >= event.start_time) || (event.end_time >= self.start_time && self.start_time >= event.start_time)
   end
+
+  private
+  # Does not allow people to fill in times enclosing other times already claimed
+  # For Example,
+  # If Person A is available from 9:00 AM to 10:00 AM,
+  # He/She may not fill in another time from 9:15 to 10:15 AM
+  # However he/she may fill in from 10:00 AM to 10:15 AM
+  def must_not_enclose_previously_claimed_times
+    if person.availabilities.where("? > end_time AND end_time > ? OR ? > start_time AND start_time > ?", end_time, start_time, end_time, start_time).present?
```

Comment:
Two things: 
Could you please break this line so it fits on Github ?
Please exclude Availabilities that are `cancelled`, since they don't count.